embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/session/BackgroundActivityDisabledTest.kt

```diff
@@ -0,0 +1,231 @@
+        }
+    }
+
+    @Test
+    fun `session span and payloads structurally correct`() {
+        with(testRule) {
+            val session1StartMs = harness.overriddenClock.now()
+            harness.overriddenClock.tick(500L)
+            val session1 = checkNotNull(harness.recordSession())
+            val session1EndMs = harness.overriddenClock.now()
+            val session2StartMs = harness.overriddenClock.tick(15000)
+            val session2 = checkNotNull(harness.recordSession())
+            val session2EndMs = harness.overriddenClock.now()
+            assertEquals(2, harness.getSentSessions().size)
+            assertEquals(0, harness.getSentBackgroundActivities().size)
+
+            assertEquals(session1.metadata, session2.metadata)
+            assertEquals(
+                session1.resource?.copy(screenResolution = null, jailbroken = null),
+                session2.resource?.copy(screenResolution = null, jailbroken = null)
+            )
+            assertEquals(session1.version, session2.version)
+            assertEquals(session1.type, session2.type)
+
+            val sessionSpan1 = session1.findSessionSpan()
+            val sessionSpan2 = session2.findSessionSpan()
+            sessionSpan1.assertExpectedSessionSpanAttributes(
+                startMs = session1StartMs,
+                endMs = session1EndMs,
+                sessionNumber = 1,
+                sequenceId = 1,
+                coldStart = true,
+            )
+
+            sessionSpan2.assertExpectedSessionSpanAttributes(
+                startMs = session2StartMs,
+                endMs = session2EndMs,
+                sessionNumber = 2,
+                sequenceId = 6,
+                coldStart = false,
+            )
+
+            assertNotEquals(
+                sessionSpan1.attributes?.findAttributeValue(SessionIncubatingAttributes.SESSION_ID.key),
+                sessionSpan2.attributes?.findAttributeValue(SessionIncubatingAttributes.SESSION_ID.key)
+            )
+
+            assertEquals(
+                sessionSpan1.attributes?.findAttributeValue(embProcessIdentifier.attributeKey.key),
+                sessionSpan2.attributes?.findAttributeValue(embProcessIdentifier.attributeKey.key)
+            )
+        }
+    }
+
+    private fun runLoggingThread() {
+        (testRule.harness.overriddenWorkerThreadModule as FakeWorkerThreadModule).executor.runCurrentlyBlocked()
+    }
+
+    private fun flushLogBatch() {
+        testRule.bootstrapper.customerLogModule.logOrchestrator.flush(false)
+    }
+
+    private fun Span.assertExpectedSessionSpanAttributes(
+        startMs: Long,
+        endMs: Long,
+        sessionNumber: Int,
+        sequenceId: Int,
+        coldStart: Boolean,
+    ) {
+        assertEquals(startMs, startTimeNanos?.nanosToMillis())
+        assertEquals(endMs, endTimeNanos?.nanosToMillis())
+        assertEquals(sessionNumber.toString(), attributes?.findAttributeValue(embSessionNumber.attributeKey.key))
```

Comment:
nit: extract to `val attrs = checkNotNull(attributes)`?

Reply:
Oh yeah sure